ledger/controllers/surveyor.js

```diff
@@ -259,6 +295,23 @@ v2.update =
 /*
    GET /v2/surveyor/{surveyorType}/{surveyorId}/{uId}
  */
+const currencyPattern = /^[0-9A-Z]{2,}$/
+const joiSurveyor = Joi.object().keys({
+  surveyorId: Joi.string().required().description('identifier for the surveyor'),
+  surveyVK: Joi.string().required().description('public key for the surveyor'),
+  registrarVK: Joi.string().required().description('public key for the associated registrar'),
+  signature: Joi.string().required().description('initialization response for the surveyor'),
+  payload: Joi.object().keys({
+    cohort: Joi.string().allow(['control', 'grant', 'test']),
+    adFree: Joi.object().keys({
+      fee: Joi.object().pattern(currencyPattern, Joi.number()).min(1).description('value after fees'),
+      choices: Joi.object().pattern(currencyPattern, Joi.array().items(Joi.number().required()).min(1)).description('currencies to choose to send'),
+      votes: Joi.number().optional().description('number of votes for distribution'),
+      altcurrency: Joi.string().regex(currencyPattern).optional().description('currency to be distributed'),
+      probi: Joi.string().regex(/\d+/).optional().description('amount of the altcurrency to be distributed')
+    }).required().description('settings for ad free mode')
+  }).required().description('additional information')
+}).required().description('response payload')
```

Comment:
can we call this surveyorValidator for consistency with https://github.com/brave-intl/bat-ledger/blob/68e21c3ad269d376fb92aed8b807099619d3aa4d/ledger/controllers/grants.js#L53-L65?

Reply:
I would prefer we undo these validator changes and just constrain this PR to the exchange rate derived contribution amount changes

Reply:
fair enough